code/script.js

```diff
@@ -218,108 +249,122 @@ const generateBoard = () => {
-}
+  secret = charactersInPlay[Math.floor(Math.random() * (charactersInPlay.length - 1 - 0 + 1) + 0)];
+  console.log('The secret person is', secret.name); 
+};
 
 // This function to start (and restart) the game
 const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
-  charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  charactersInPlay = CHARACTERS // Here we're setting charactersInPlay array to be all the characters to start with
+  generateBoard();
+  setSecret();
+  selectQuestion();
+  winOrLose.style.display = 'none'; //This will hide the winOrLose block for when restarting the game.
+  //If i want a time function, set it here
 }
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
-
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  
+  const category = questions.options[questions.selectedIndex].parentNode.label // This variable stores what option group (category) the question belongs to.
 
+  const value = questions.options[questions.selectedIndex].value   // This variable stores the acctual value of the question we've selected.
+  //currentQuestion is "Does the person have "category, value"?"
   currentQuestion = {
-    category: category,
-    // value: value
+    category: category, // Based on the option group
+    value: value,       // Comes from the selected option
   }
 }
 
+// This was tricky to understand, but this site was helpful. https://stackoverflow.com/c/technigo/questions/2884
 // This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
-
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
+  const { category, value } = currentQuestion;
+  let keep = false; //if keep is true in conditions below, only they will be filtered. https://stackoverflow.com/c/technigo/questions/3526?rq=1
+  console.log('Guess on',category, value);
+  console.log('This is', secret[category].includes(value)); // Let us know if true or false
+  
+  // Hair, eyes and gender are strings
+  if (category === 'hair' || category === 'eyes'|| category === 'gender') {
+    if (secret[category] === value) {
+      keep = true
+    } 
+    // Accessories and other are arrays
   } else if (category === 'accessories' || category === 'other') {
-
+    if (secret[category].includes(value)) {
+      keep = true
+    }
```

Comment:
Nice and clean code, great that you separated the strings and arrays. Helpful console logs.